pandas/core/internals/construction.py

```diff
@@ -107,7 +111,8 @@ def arrays_to_mgr(
 
         # don't force copy because getting jammed in an ndarray anyway
         arrays = _homogenize(arrays, index, dtype)
-
+        if typ == "array":
+            arrays = [ensure_wrapped_if_datetimelike(arr) for arr in arrays]
```

Comment:
could do this unconditionally.  tradeoff: slightly less complex code here, perf of unecessary wrapping/unwrapping in DatetimeBlock/TimedeltaBlock._maybe_coerce_values (for now, until upcoming PR)

Reply:
I would then leave that until it is useful

Reply:
can you do this *inside* e.g. in ArrayManager constructor (or add a constructor on L127), prob should do that anyhow.

Reply:
> add a constructor on L127

What do you mean exactly with that? 
We already use the main constructor `ArrayManager(arrays, [index, columns])` on L127 because that constructor already works from "arrays" (in constrast to the BlockManager, which needs to consolidate arrays etc, so therefore there is a helper `create_blockmanager_from_arrays`, but for ArrayManager this is exactly the `__init__` already)

Reply:
i think the suggestion is to move the call on L115 to just before the `return ArrayManager` on L127.  i think thatd be a small improvement, not a big deal

Reply:
I intentionally put it here because, in theory, I think this wrapping only needs to be done with the default of `verify_integrity=True` (so you have a way to skip it)

Reply:
I mean on line L127 this should call a method *inside* internal and not directly calling the manager. Then you can handle the special cases. on L110 *inside* that function.

Reply:
So you want me to write this function?

```
def create_array_manager_from_arrays(arrays, axes, verify_integrity=True):
    if verify_integrity:
        arrays = [ensure_wrapped_if_datetimelike(arr) for arr in arrays]
    return ArrayManaget(arrays, axes)
```

(IMO this just splits the handling of `verify_integrity` to multiple places (as mentioned above, that's the reason that I put this wrapping where it is located now), but I don't care if this makes you happy)

>  this should call a method *inside* internal 

Note you are commenting on a file that *already* is *inside* internals

Reply:
its L110 which needs to be handled in array_manager.py and not here

Reply:
OK, moved inside `array_manager.py`